lib/formatting.js

```diff
@@ -0,0 +1,47 @@
+import ircFormatting from 'irc-formatting';
+import SimpleMarkdown from 'simple-markdown';
+import colors from 'irc-colors';
+import _ from 'lodash';
+
+function mdNodeToIRC(node) {
+  let content = node.content;
+  if (_.isArray(content)) content = content.map(mdNodeToIRC).join('');
+  if (node.type === 'em') return colors.italic(content);
+  if (node.type === 'strong') return colors.bold(content);
+  if (node.type === 'u') return colors.underline(content);
+  return content;
+}
+
+export function formatFromDiscordToIRC(text) {
+  const markdownAST = SimpleMarkdown.defaultInlineParse(text);
+  return markdownAST.map(mdNodeToIRC).join('');
+}
+
+export function formatFromIRCToDiscord(text) {
+  const blocks = ircFormatting.parse(text).map(block =>
+    // Consider reverse as italic, some IRC clients use that
+    _.assign({}, block, { italic: block.italic || block.reverse })
```

Comment:
The package has a Node >6 requirement anyway, since discord.js enforces that, so could do
```js
{
  ...block,
  { italic: block.italic || block.reverse }
}
```